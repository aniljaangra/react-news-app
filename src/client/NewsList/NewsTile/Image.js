/**
 * Created by Anil Jangra on 5/23/2021
 */
import React, { useState } from 'react';

const Image = ({ src }) => {
  const [isLoading, setIsLoading] = useState(true);

  function onLoad() {
    setIsLoading(false);
  }

  return (
    <>
      <img
        className="card-img-top"
        src="https://via.placeholder.com/300x200/fff"
        style={{ display: isLoading ? 'block' : 'none' }}
        alt="IMG"
      />
      <img
        className="card-img-top"
        src={src}
        style={{ display: isLoading ? 'none' : 'block' }}
        onLoad={onLoad}
        alt="IMG"
      />
    </>
  );
};

export default Image;
