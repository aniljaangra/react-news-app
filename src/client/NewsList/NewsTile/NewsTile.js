import React from 'react';
import Image from './Image';

function NewsTile({ news }) {
  return (
    <div className="col-lg-4 col-md-6 col-sm-12 d-flex align-items-stretch">
      <div
        className="card shadow mb-5 p-2 bg-white rounded"
      >
        <Image src={news.urlToImage} />
        <div className="card-body">
          <h5 className="card-title">
            {news.title}
          </h5>
          <p className="card-text">
            {news.description}
          </p>
        </div>
        <a
          href={news.url}
          className="btn btn-primary"
          target="_blank"
          style={{ width: '50%', margin: '5% auto 10% auto' }}
          rel="noreferrer"
        >
          {news.source.name}
        </a>
      </div>
    </div>
  );
}
export default NewsTile;
