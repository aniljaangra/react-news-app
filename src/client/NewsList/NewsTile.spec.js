/**
 * Created by Anil Jangra on 5/24/2021
 */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewsTile from './NewsTile/NewsTile';

describe('NewsTile', () => {
  const news = {
    title: 'sample news',
    description: 'sample description',
    url: 'https://localhost',
    urlToImage: 'http://localhost/image',
    source: {
      name: 'sample source',
    },
  };
  it('renders NewsTile component correctly', () => {
    const tree = shallow(<NewsTile news={news} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
