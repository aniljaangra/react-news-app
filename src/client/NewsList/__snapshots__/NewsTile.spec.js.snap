// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsTile renders NewsTile component correctly 1`] = `
<div
  className="col-lg-4 col-md-6 col-sm-12 d-flex align-items-stretch"
>
  <div
    className="card shadow mb-5 p-2 bg-white rounded"
  >
    <Image
      src="http://localhost/image"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        sample news
      </h5>
      <p
        className="card-text"
      >
        sample description
      </p>
    </div>
    <a
      className="btn btn-primary"
      href="https://localhost"
      rel="noreferrer"
      style={
        Object {
          "margin": "5% auto 10% auto",
          "width": "50%",
        }
      }
      target="_blank"
    >
      sample source
    </a>
  </div>
</div>
`;
